File: code/2025/03/30/2109. 向字符串添加空格.cpp
class Solution {
public:
	string addSpaces(string s, vector<int>& spaces) {
		std::string ans {};
		for (int i = 0, j = 0, n = s.size(); i < n; i++) {
			if (j < spaces.size() and i == spaces[j]) {
				ans += ' ';
				j++;
			}
			ans += s[i];
		}
		return ans;
	}
};

File: code/2025/03/30/2109. 向字符串添加空格.py
class Solution:
    def addSpaces(self, s: str, sp: List[int]) -> str:
        i, j = 0, 0
        ans = ""
        for c in s:
            if i < len(sp) and sp[i] == j:
                ans += " "
                i += 1
            ans += c
            j += 1
        return ans